### ISSUE TITLE
warning in DirectoryLookup.h about bit field DirCharacteristic not wide enough to fit enum CharacteristicKind

### ISSUE THREAD

**Issue by doxygen** [llvm/llvm-project#59840]
When using LLVM 15.0.6 and including `clang/Lex/DirectoryLookup.h` with option `-Wbitfield-enum-conversion` enabled I get the following warning:

```console
DirectoryLookup.h:77:17: warning: 
      bit-field 'DirCharacteristic' is not wide enough to store all enumerators of
      'CharacteristicKind' [-Wbitfield-enum-conversion]
      : u(Map), DirCharacteristic(DT), LookupType(LT_HeaderMap),
```

`DirCharacteristic` is a bitfield with 2 bits (4 values)

```c++
  /// DirCharacteristic - The type of directory this is: this is an instance of
  /// SrcMgr::CharacteristicKind.
  unsigned DirCharacteristic : 2;
```

Whereas `SrcMgr::CharacterKind` is an enum with 5 values:
```c++
enum CharacteristicKind {
  C_User,
  C_System,
  C_ExternCSystem,
  C_User_ModuleMap,
  C_System_ModuleMap
};
```
Probable solution: increase `DirCharacteristic` bitfield from 2 to 3.

**Comment by xgupta** [comment]
I uploaded a patch on your behalf  - https://reviews.llvm.org/D142304.

**Comment by AaronBallman** [comment]
Fixed in 94b9187a7d37e4269af35f8f7ec8e0f78fd6a06e
